experiments/tf_trainer/common/text_preprocessor.py

```diff
@@ -19,26 +34,27 @@
                          'Enable fine tuning of embeddings.')
 
 class TextPreprocessor(object):
-  """Text Preprocessor.
+  """Text Preprocessor TensorFlow Estimator Extension.
 
-  Takes an embedding and uses it to produce a word to index mapping and an
-  embedding matrix.
+  Uses embedding indexes to create tensors that map tokens (provided by an
+  abstract tokenizer funtion) to embeddings.
 
-  Note: Due to the lack of text preprocessing functions in tensorflow, we expect 
-    that the text is already preprocessed (list of words) in inference.
-    In training, due to the availability of tf.py_func, we handle the preprocessing.
+  Note: Due to the lack of text preprocessing functions in tensorflow, we expect
+  that the text is already preprocessed (list of words) in inference. In
+  training, due to the availability of tf.py_func, we can handle the
+  preprocessing.
+
+  The reasoning for this comes from this observation: The embedding step can be
```

Comment:
I am not sure to understand.  Is "Word to index" the vocabulary as word --> integer? why is this the limiting factor and how can the max word length solve the issue?

Also, is it related to the comment above ("due to the lack...")? The problem I had was related to the string preprocessing and in particular the tokenization.

Reply:
good question, I think this is an old bit of text, and is more confusing and not very helpful compared to the comment above. So I'm removing for now. :)